src/ProductConstructionService/ProductConstructionService.ReproTool/ReproTool.cs

```diff
@@ -0,0 +1,280 @@
+        }
+        // Backward flow
+        else
+        {
+            vmrBranch = defaultChannel.Branch;
+            productRepoUri = subscription.TargetRepository;
+            productRepoBranch = subscription.TargetBranch;
+            isForwardFlow = false;
+        }
+        var productRepoForkUri = ProductRepoFormat + productRepoUri.Split('/').Last();
+
+        await using var vmrTmpBranch = await PrepareVmrForkAsync(vmrBranch, productRepoUri, productRepoForkUri);
+
+        await PrepareProductRepoForkAsync(productRepoUri, productRepoForkUri, productRepoBranch);
+
+        // Find the latest commit in the source repo to create a build from
+        string sourceRepoSha;
+        (string sourceRepoOwner, string sourceRepoName) = GetRepoOwnerAndName(subscription.SourceRepository);
+        if (string.IsNullOrEmpty(options.Commit))
+        {
+            var res = await ghClient.Git.Reference.Get(sourceRepoOwner, sourceRepoName, $"heads/{defaultChannel.Branch}");
+            sourceRepoSha = res.Object.Sha;
+        }
+        else
+        {
+            // Validate that the commit actually exists
+            try
+            {
+                await ghClient.Repository.Commit.Get(sourceRepoOwner, sourceRepoName, options.Commit);
+            }
+            catch (NotFoundException)
+            {
+                throw new ArgumentException($"Commit {options.Commit} doesn't exist in repo {subscription.SourceRepository}");
+            }
+            sourceRepoSha = options.Commit;
+        }
+
+        // Create the temp channel
+        var channelName = Guid.NewGuid().ToString();
+        await using var channel = await darcProcessManager.CreateTestChannelAsync(channelName);
+
+        // The build
+        var build = await CreateBuildAsync(
+            isForwardFlow ? productRepoForkUri : VmrForkUri,
+            isForwardFlow ? defaultChannel.Branch : vmrTmpBranch.Value,
+            sourceRepoSha);
+
+        // And the subscription
+        await using var testSubscription = await darcProcessManager.CreateSubscriptionAsync(
+            channel: channelName,
+            sourceRepo: isForwardFlow ? productRepoForkUri : VmrForkUri,
+            targetRepo: isForwardFlow ? VmrForkUri : productRepoForkUri,
+            targetBranch: isForwardFlow ? vmrTmpBranch.Value : subscription.TargetBranch,
+            sourceDirectory: subscription.SourceDirectory,
+            targetDirectory: subscription.TargetDirectory);
+
+        // Add it to the channel
+        await darcProcessManager.AddBuildToChannelAsync(build.Id, channelName);
+
+        // Trigger the subscription
+        await TriggerSubscriptionAsync(testSubscription.Value);
+
+        logger.LogInformation("Code flow successfully recreated. Press enter to finish and cleanup");
+        Console.ReadLine();
+
+        // Cleanup
+        await DeleteAllDarcPRBranchesAsync(isForwardFlow ? VmrForkRepoName : productRepoUri.Split('/').Last());
```

Comment:
Isn't this too much? What if more people are testing something?

Reply:
yeah, definitely too much. Trying to figure out the branch name format in backwards flow cases

Reply:
ah it's the same, just darc-<target branch name>-<some guid>. I'll make the change to create a tmp branch in the product repo fork too in case of backwards flow so we don't delete all branches

Reply:
we'll create tmp branches in the product repo for now too (always). Also added logic so only the PR branch from this test is deleted, so we won't be disruptions